src/expr/src/scalar/mod.rs

```diff
@@ -742,6 +742,53 @@ impl MirScalarExpr {
                         {
                             // Canonically order elements so that deduplication works better.
                             mem::swap(expr1, expr2);
+                        } else if let (
+                            BinaryFunc::Eq,
+                            MirScalarExpr::Literal(
+                                Ok(lit_row),
+                                ColumnType {
+                                    scalar_type:
+                                        ScalarType::Record {
+                                            fields: field_types,
+                                            ..
+                                        },
+                                    ..
+                                },
+                            ),
+                            MirScalarExpr::CallVariadic {
+                                func: VariadicFunc::RecordCreate { .. },
+                                exprs: rec_create_args,
+                            },
+                        ) = (func, (**expr1).clone(), (**expr2).clone())
+                        {
+                            // Literal(record(c1, c2)) = record_create(e1, e2)
```

Comment:
nit:
```
// Literal(record(c1, c2)) = record_create(e1, e2)
```

strictly speaking the literal value is a list, so I would write it as `Literal([c1, c2])`.

Reply:
Fixed. I also added a comment to explain that we represent records as lists, because this confused me a bit at first.